lib/flipper/api/v1/actions/groups_gate.rb

```diff
@@ -27,7 +27,23 @@ def delete
           private
 
           def ensure_valid_params
-            json_error_response(:group_not_registered) unless Flipper.group_exists?(group_name)
+            if group_name.nil? || group_name.empty?
+              json_error_response(:name_invalid)
+            end
+
+            return if allow_unregistered_groups?
+            return if Flipper.group_exists?(group_name)
+
+            json_error_response(:group_not_registered)
+          end
+
+          def allow_unregistered_groups?
+            allow_unregistered_groups = params['allow_unregistered_groups']
```

Comment:
cool, I really like this approach and saw that the adapters implements this now too from #245 ! I agree that specific parameters is a much better idea.  I think we still  need to do something similar for the preloading in `features#get` right (so that the adapter can pass a parameter to get all requested features even if they don't exist (avoiding unnecessary queries)?